packages/contentstack-import/src/lib/import/custom-roles.js

```diff
@@ -0,0 +1,80 @@
+'use strict';
+
+const mkdirp = require('mkdirp');
+const fs = require('fs');
+const path = require('path');
+const chalk = require('chalk');
+
+const helper = require('../util/fs');
+const { addlogs } = require('../util/log');
+const stack = require('../util/contentstack-management-sdk');
+let config = require('../../config/default');
+
+const customRolesConfig = config.modules.customRoles;
+let customRolesFolderPath;
+let customRolesMapperPath;
+let customRolesUidMapperPath;
+let customRolesSuccessPath;
+let customRolesFailsPath;
+
+function ImportCustomRoles() {
+  this.fails = [];
+  this.customRolesUidMapper = {};
+  this.labelUids = [];
+  this.client = null;
+  if (fs.existsSync(customRolesMapperPath)) {
+    this.customRolesUidMapper = helper.readFile(customRolesUidMapperPath) || {};
+  }
+}
+
+ImportCustomRoles.prototype.start = async function(credentialConfig) {
+  let self = this;
+  try {
+    config = credentialConfig;
+    this.client = stack.Client(config);
+    addlogs(config, chalk.white('Migrating custom-roles'), 'success');
+    customRolesFolderPath = path.resolve(config.data, customRolesConfig.dirName);
+    self.customRoles = helper.readFile(path.resolve(customRolesFolderPath, customRolesConfig.fileName));
+    customRolesMapperPath = path.resolve(config.data, 'mapper', 'custom-roles');
+    customRolesUidMapperPath = path.resolve(config.data, 'mapper', 'custom-roles', 'uid-mapping.json');
+    customRolesSuccessPath = path.resolve(config.data, 'custom-roles', 'success.json');
+    customRolesFailsPath = path.resolve(config.data, 'custom-roles', 'fails.json');
+    mkdirp.sync(customRolesMapperPath);
+
+    if (!self.customRoles) {
+      addlogs(config, chalk.white('No custom-roles found'), 'error');
+      return;
+    }
+    self.customRolesUids = Object.keys(self.customRoles);
+    for (const uid of self.customRolesUids) {
+      const customRole = self.customRoles[uid];
+
+      if (uid in self.customRolesUidMapper) {
```

Comment:
do we to loop in mapper to check it already created or not, or  self.customRolesUidMapper[uid]  o(1) operation

Reply:
@shafeeqd959 There is no loop in mapper object. Looping is done for keys in customRoles object.